hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBBatchOperation.java

```diff
@@ -127,12 +131,21 @@ public int hashCode() {
 
     @Override
     public String toString() {
-      return array != null ? bytes2String(array)
-          : bytes2String(asReadOnlyByteBuffer());
+      return slice.toString();
```

Comment:
Have you checked what will it return?  Could you show an example output?

Reply:
```
public static void main(String[] args) throws CodecException {
    String value = "test";
    Bytes bytes = new Bytes(value.getBytes(UTF_8));
    System.out.println("To String Heap Value: " + bytes);
    Bytes dbytes = newBytes(CodecBufferCodec.get(true).fromPersistedFormat(value.getBytes(UTF_8)));
    System.out.println("To String Direct Value: " + dbytes);
  }
```

```
2025-12-28 18:25:34,786 [main] INFO  db.CodecBuffer (CodecBuffer.java:set(85)) - Successfully set constructor to LeakDetector::newCodecBuffer: org.apache.hadoop.hdds.utils.db.CodecBuffer$$Lambda$4/1694556038@7085bdee
To String Heap Value: test
To String Direct Value: test
```